File: PDP Second Project/src/BUretim.c
#include "BUretim.h"

BUretim BUretimOlustur()
{
    BUretim this;
    this = (BUretim)malloc(sizeof(struct BURETIM));
    this->super = UretimOlustur();
    this->super->Uret = &UretB;    
    this->UretimPopulasyonAl = &BUretimPopulasyonAl; 
    this->UretimYoket = &BUretimYoket;

    return this;
}

int UretB(const BUretim this)
{
    return (int)pow(this->super->pop,1/3) % 10;
}

void BUretimPopulasyonAl(const BUretim this, int pop)
{
	this->super->UretimPopulasyonAl(this->super, pop);
}

void BUretimYoket(const BUretim this){
	if(this == NULL) return;
	this->super->UretimYoket(this->super);
	free(this);
}

File: PDP_3rd_Project/src/BUretim.java

public class BUretim implements IUretim{

	private int pop = 0;
	@Override
	public void UretimPopulasyonAl(int pop) {
		
		this.setPop(pop);
	}

	@Override
	public int Uret() {
		
		return (int) Math.pow(pop, 1/3) % 10;
	}

	public void setPop(int pop) {
		this.pop = pop;
	}
}
